integration_tests/test_store_register.py

```diff
@@ -80,9 +80,11 @@ def test_registrations_in_a_row_fail_if_too_fast(self):
 
         self.register(snap_name_1, wait=False)
 
+        # Wait after the registration attempt, so the following registrations
+        # don't get the error.
         error = self.assertRaises(
             subprocess.CalledProcessError,
-            self.register, snap_name_2, wait=False)
+            self.register, snap_name_2, wait=True)
```

Comment:
this won't wait as the exception would be raised. We either need to catch and reraise in `<class>.register` or wait and the end of the test here (`addCleanup`)